fish-rust/src/io.rs

```diff
@@ -189,8 +194,29 @@ pub trait IoData {
     fn print(&self);
     // The address of the object, for comparison.
     fn as_ptr(&self) -> *const ();
+    fn as_bufferfill(&self) -> Option<&IoBufferfill> {
+        None
+    }
 }
 
+// todo!("later: this should be safe because of how it's used. Rationalize this better.")
+pub trait IoDataSync: IoData + Send + Sync {}
+unsafe impl Send for IoClose {}
```

Comment:
These are a bit ugly. I'm pretty sure there is no problem in practice but I didn't check yet.